nextcord/user.py

```diff
@@ -143,8 +147,11 @@ def default_avatar(self) -> Asset:
         """:class:`Asset`: Returns the default avatar for a given user.
 
         This is calculated by the user's discriminator.
+
+        ..versionchanged:: 2.5
+            Indexes avatar using ID instead of discriminator.
         """
-        return Asset._from_default_avatar(self._state, int(self.discriminator) % len(DefaultAvatar))
+        return Asset._from_default_avatar(self._state, (self.id >> 6) % len(DefaultAvatar))
```

Comment:
This should check for `'0'` for now so non-migrated users are still correct. Something like this
```suggestion
        if self.discriminator == '0':
            avatar_id = (self.id >> 22) % len(DefaultAvatar)
        else:
            avatar_id = int(self.discriminator) % 5

        return Asset._from_default_avatar(self._state, avatar_id)
```

> ** In the case of the Default User Avatar endpoint, the value for `index` depends on whether the user is [migrated to the new username system](https://discord.com/developers/docs/change-log#unique-usernames-on-discord). For users on the new username system, `index` will be `(user_id >> 22) % 6`. For users on the legacy username system, `index` will be `discriminator % 5`.